cmd/juju/model/destroy.go

```diff
@@ -200,6 +212,37 @@ func (c *destroyCommand) getStorageAPI() (StorageAPI, error) {
 	return storage.NewClient(root), nil
 }
 
+// getMachineIds gets slice of machine ids from modelData.
+func getMachineIds(data base.ModelStatus) []string {
+	return transform.Slice(data.Machines, func(f base.Machine) string {
+		return f.Id + " (\"" + f.InstanceId + "\")"
+	})
+}
+
+// printDestroyWarning prints to stdout the warning with additional info about destroying model.
+func printDestroyWarning(ctx *cmd.Context, api DestroyModelAPI, modelName string, modelDetails *jujuclient.ModelDetails) error {
+	modelStatuses, err := api.ModelStatus(names.NewModelTag(modelDetails.ModelUUID))
+	if err != nil {
+		return errors.Annotate(err, "getting model status")
+	}
+	machineIds := getMachineIds(modelStatuses[0])
+	msg := destroyModelMsg
+	if len(machineIds) > 0 {
+		msg = destroyIAASModelMsgWithDetails
+		if modelDetails.ModelType == model.CAAS {
+			msg = destroyCAASModelMsgWithDetails
+		}
+		fmt.Fprintf(ctx.Stdout, msg, modelName,
+			modelStatuses[0].TotalMachineCount,
+			modelStatuses[0].ApplicationCount,
+			strings.Join(machineIds, ", "),
+		)
+	} else {
+		fmt.Fprintf(ctx.Stdout, msg, modelName)
```

Comment:
I prefer to be explicit about ignoring returns for these, so `_, _ = fmt.Fprintf(...)`